lib/rex/socket.rb

```diff
@@ -157,7 +157,27 @@ def self.dotted_ip?(addr)
     (support_ipv6? && addr =~ MATCH_IPV6) || (addr =~ MATCH_IPV4)
   end
 
-  #
+  # Checks to see if an address is an IPv6 address and if so, converts it into its
+  # square bracket format for addressing as noted in RFC 6874 which states that an IPv6
+  # address literal in a URL is always embedded between [ and ]. Please also refer to
+  # RFC5952 and RFC3986 for more info.
+  #
+  # RFC3986 specifically notes "A host identified by an Internet Protocol literal address, version 6
+  # [RFC3513] or later, is distinguished by enclosing the IP literal
+  # within square brackets ("[" and "]").  This is the only place where
+  # square bracket characters are allowed in the URI syntax."
+  #
+  # @param addr [String] IPv6 address to convert to URL format.
+  # @return [String, nil] Returns the converted address on success, or nil on failure.
+  def self.convert_ipv6_to_url_fmt(addr)
+    if is_ipv6?(addr)
+      "[#{addr}]"
+    else
+      nil
+    end
+  end
+
```

Comment:
This makes a few changes:

1. Renames it to `.to_authority` (see [RFC3986 Section 3](https://datatracker.ietf.org/doc/html/rfc3986#section-3)). Alternatively you could call it `.to_uri_authority` I guess to keep it inline with the RFC3986 verbiage and be more explicit but then that's 4 more character to type and the idea here is to make this more convenient.
2. Takes the port because it's only necessary to enclose the address with brackets when a port is included. If the port is `Nil` the host is returned as is, again for the caller's convenience.
3. Always returns a string because this will make it easier to use. The caller won't have to check if the conversion took place, they can simply use the result.

```suggestion
  # Checks to see if an address is an IPv6 address and if so, converts it into its
  # square bracket format for addressing as noted in RFC 6874 which states that an IPv6
  # address literal in a URL is always embedded between [ and ]. Please also refer to
  # RFC5952 and RFC3986 for more info.
  #
  # RFC3986 section 3.2.2 specifically notes "A host identified by an Internet Protocol literal address, version 6
  # [RFC3513] or later, is distinguished by enclosing the IP literal
  # within square brackets ("[" and "]").  This is the only place where
  # square bracket characters are allowed in the URI syntax."
  #
  # @param host [String] IP address or hostname to convert to a URI authority.
  # @param port [Integer] Port number to include within the URI authority.
  # @return [String] Returns the URI authority string.
  def self.to_authority(host, port=nil)
    host = "[#{host}]" if port && is_ipv6?(host)
    host += ":#{port}" if port
    host
  end
```

Reply:
Suggested change implemented in https://github.com/rapid7/rex-socket/pull/61/commits/07ebf46746ac39e0f1079c9af2f71ee9a608deea.